RetailCoder.VBE/Inspections/QuickFixes/AssignedByValParameterMakeLocalCopyQuickFix.cs

```diff
@@ -97,13 +97,42 @@ private void ReplaceAssignedByValParameterReferences(IModuleRewriter rewriter, s
             }
         }
 
+
         private void InsertLocalVariableDeclarationAndAssignment(IModuleRewriter rewriter, string localIdentifier)
         {
-            var content = Tokens.Dim + " " + localIdentifier + " " + Tokens.As + " " + _target.AsTypeName + Environment.NewLine
-                + (_target.AsTypeDeclaration is ClassModuleDeclaration ? Tokens.Set + " " : string.Empty)
-                + localIdentifier + " = " + _target.IdentifierName;
+            var content = Tokens.Dim + " " + localIdentifier + " " + Tokens.As + " " + _target.AsTypeName + Environment.NewLine;
+            if (IsBaseTypeContext(_target))
+            {
+                content = content + localIdentifier + " = " + _target.IdentifierName;
+            }
+            else 
+            {
+                //All we can know is that it is not a Base type.  Let VBA determine
+                //the right way to assign the parameter.  The user can simplify it later.
+                string insertIsObjectCheck =
+@"If(IsObject({1})) Then
```

Comment:
Haven't checked the code, so just throwing out a couple things - You can check the `IsTypeSpecified` property to see if it's implicitly `Variant`, and the `AsTypeName` to see if it's explicitly `Variant`.  Beyond that, I'm not sure that there is much that you can do to determine whether or not `Set` is required until we have better code path analysis because the RHS of the assignment can be an expression of arbitrary complexity. Consider this:

```
Public Sub Foo(ByVal arg1)
   Dim bar As Variant
   Set bar = New Collection
   arg1 = bar
End Sub
```
Now rinse and repeat as many times as you want - in the example above, all we know ATM is that `bar` is another `Variant`.  Maybe in another case `bar` is assigned by another function that returns a `Variant`. I'd say that for now these should just have the changes previewed and trust the user to make the right call.